oneandone/src/main/java/org/apache/jclouds/oneandone/rest/compute/function/ServerToNodeMetadata.java

```diff
@@ -78,26 +81,43 @@ public ServerToNodeMetadata(Function<Hdd, Volume> fnVolume,
    @Override
    public NodeMetadata apply(final Server server) {
       checkNotNull(server, "Null server");
+      String hardwareId = null;
 
       DataCenter dataCenter = api.dataCenterApi().get(server.datacenter().id());
       Location location = find(locations.get(), idEquals(dataCenter.id()));
```

Comment:
you already have the `datacenter.id` you don't need to call the api to get the datacenter object

Reply:
changed